openssl-dynamic/src/main/c/ocsp.c

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "ocsp.h"
+#include "tcn.h"
+#include "ssl_private.h"
+
+#include <stdio.h>
+
+#ifndef TCN_NO_OCSP
+#ifndef OPENSSL_IS_BORINGSSL
+
+#ifndef TCN_OCSP_CLIENT_CALLBACK_ERROR
```

Comment:
it doesn't look like these defines are used outside this scope ... if these are already defined wouldn't this be a programming error we want to investigate? Is the intention to allow overriding this as part of the build environment?